scipy/sparse/sparsetools/csc.h

```diff
@@ -30,23 +33,44 @@ void csc_matvec(const I n_row,
 	            const I Ap[], 
 	            const I Ai[], 
 	            const T Ax[],
+                    const T alpha,
 	            const T Xx[],
+                    const T beta,
 	                  T Yx[])
 { 
-    for(I j = 0; j < n_col; j++){
-        I col_start = Ap[j];
-        I col_end   = Ap[j+1];
+    scal(n_row, beta, Yx);
 
-        for(I ii = col_start; ii < col_end; ii++){
-            I i    = Ai[ii];
-            Yx[i] += Ax[ii] * Xx[j];
+    if (alpha == 0) {
+        return;
+    }
+    else if (alpha == 1) {
+        for(I j = 0; j < n_col; j++){
+            I col_start = Ap[j];
+            I col_end   = Ap[j+1];
+
+            for(I ii = col_start; ii < col_end; ii++){
+                I i    = Ai[ii];
+                Yx[i] += Ax[ii] * Xx[j];
+            }
+        }
+    }
+    else {
+        for(I j = 0; j < n_col; j++){
+            I col_start = Ap[j];
+            I col_end   = Ap[j+1];
+
+            for(I ii = col_start; ii < col_end; ii++){
+                I i    = Ai[ii];
+                Yx[i] += alpha *  Ax[ii] * Xx[j];
+            }
```

Comment:
Are you sure the overhead of a single multiplication with a scalar (a register operation, presumably) warrants this duplication?